wgsl/index.bs

```diff
@@ -3142,36 +3109,451 @@ TODO: deduplicate these tables
+    <td>(GLSLstd450SmoothStep)
+  <tr algorithm="scalar case, sqrt">
+    <td>|T| is f32
+    <td class="nowrap">`sqrt(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Sqrt)
+  <tr algorithm="vector case, sqrt">
+    <td>|T| is f32
+    <td class="nowrap">`sqrt(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Sqrt)
+  <tr algorithm="scalar case, step">
+    <td>|T| is f32
+    <td class="nowrap">`step(`|e1|`:` |T| `, `|e2|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Step)
+  <tr algorithm="vector case, step">
+    <td>|T| is f32
+    <td class="nowrap">`step(`|e1|`:` vec|N|<|T|> `, `|e2|`:` vec|N|<|T|>`) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Step)
+  <tr algorithm="scalar case, tan">
+    <td>|T| is f32
+    <td class="nowrap">`tan(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Tan)
+  <tr algorithm="vector case, tan">
+    <td>|T| is f32
+    <td class="nowrap">`tan(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Tan)
+  <tr algorithm="scalar case, tanh">
+    <td>|T| is f32
+    <td class="nowrap">`tanh(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Tanh)
+  <tr algorithm="vector case, tanh">
+    <td>|T| is f32
+    <td class="nowrap">`tanh(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Tanh)
+  <tr algorithm="scalar case, trunc">
+    <td>|T| is f32
+    <td class="nowrap">`trunc(`|e|`:` |T| `) -> ` |T|
+    <td>(GLSLstd450Trunc)
+  <tr algorithm="vector case, trunc">
+    <td>|T| is f32
+    <td class="nowrap">`trunc(`|e|`:` vec|N|<|T|> `) -> ` vec|N|<|T|>
+    <td>(GLSLstd450Trunc)
+</table>
+
 ## Integer built-in functions ## {#integer-builtin-functions}
 
 <table class='data'>
   <thead>
     <tr><td>Precondition<td>Built-in<td>Description
   </thead>
-  <tr algorithm="scalar case, count 1 bits"><td>
-          |T| is u32 or i32<br>
-      <td class="nowrap">`countOneBits(`|e|`:` |T| `) ->` |T|
-      <td>The number of 1 bits in the representation of |e|.<br>
-          Also known as "population count".<br>
-          (SPIR-V OpBitCount)
-  <tr algorithm="vector case, count 1 bits"><td>
-          |T| is u32 or i32
-      <td class="nowrap">`countOneBits(`|e|`:` vec|N|<|T|>`) ->` vec|N|<|T|><br>
-      <td>Component-wise population count:
-          Component |i| of the result is `countOneBits(`|e|`[`|i|`])`<br>
-          (SPIR-V OpBitCount)
-  <tr algorithm="scalar bit reversal"><td>
-          |T| is u32 or i32<br>
-      <td class="nowrap">`reverseBits(`|e|`:` |T| `) ->`  |T|
-      <td>Reverses the bits in |e|:  The bit at position |k| of the result equals the
-          bit at position 31-|k| of |e|.<br>
-          (SPIR-V OpBitReverse)
-  <tr algorithm="vector bit reversal"><td>
-          |T| is u32 or i32
-      <td class="nowrap">`reverseBits(`|e|`:` vec|N|<|T|> `) ->` vec|N|<|T|><br>
-      <td>Component-wise bit reversal:
-          Component |i| of the result is `reverseBits(`|e|`[`|i|`])`<br>
-          (SPIR-V OpBitReverse)
+  <tr algorithm="scalar case, abs">
+    <td>|T| is u32 or i32
```

Comment:
do we really need it on `u32`?

Reply:
Probably not needed, but probably doesn't hurt if it's supported by the native languages.